mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/binarystorage/GridFSDelegator.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.binarystorage;
+
+import java.io.ByteArrayOutputStream;
+
+import org.hibernate.engine.jdbc.BinaryStream;
+import org.hibernate.ogm.datastore.mongodb.options.impl.GridFSBucketOption;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.options.spi.OptionsContext;
+
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+
+/**
+ * The implementation provides features for GridFS usage
+ *
+ * @see <a href ="https://docs.mongodb.com/manual/core/gridfs">GridFS documentation</a>
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
```

Comment:
I remember asking you to add some javadoc to explain what GridFS is about but you only added a link.

Reply:
In particular you could mention why we need to deal with the additional field upload_id in the snapshot type.
Remember, the page you linked might not exist anymore 1 year from now.